instant-xml-macros/src/lib.rs

```diff
@@ -55,17 +65,29 @@ struct FieldMeta {
 }
 
 impl FieldMeta {
-    fn from_field(input: &syn::Field) -> FieldMeta {
+    fn from_field(input: &syn::Field) -> Result<FieldMeta, syn::Error> {
         let mut meta = FieldMeta::default();
         for item in meta_items(&input.attrs) {
             match item {
                 MetaItem::Attribute => meta.attribute = true,
                 MetaItem::Ns(ns) => meta.ns = ns,
                 MetaItem::Rename(lit) => meta.rename = Some(lit),
-                MetaItem::Scalar => panic!("attribute 'scalar' invalid in field xml attribute"),
+                MetaItem::RenameAll(_) => {
+                    return Err(syn::Error::new(
+                        input.span(),
+                        "attribute 'rename_all' invalid in field xml attribute",
+                    ))
+                }
+                MetaItem::Scalar => {
+                    return Err(syn::Error::new(
+                        input.span(),
```

Comment:
And here.

Reply:
See above, I think we should be consistent with how reporting happens.